shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java

```diff
@@ -0,0 +1,256 @@
+        Assert.assertEquals("host1", metricsConfiguration.getHost());
+        Assert.assertEquals(111, metricsConfiguration.getPort());
+        Assert.assertEquals(true, metricsConfiguration.getAsync());
+        Assert.assertEquals(true, metricsConfiguration.getEnable());
+        Assert.assertEquals(4, metricsConfiguration.getThreadCount());
+        Properties metricsProperties = metricsConfiguration.getProps();
+        Assert.assertNotNull(metricsProperties);
+        Assert.assertEquals(1, metricsProperties.size());
+        Assert.assertEquals("value3", metricsProperties.getProperty("key3"));
+
+        //test for props
+        Properties proxyConfigurationProps = proxyConfiguration.getProps();
+        Assert.assertNotNull(proxyConfigurationProps);
+        Assert.assertEquals(1, proxyConfigurationProps.size());
+        Assert.assertEquals("value4", proxyConfigurationProps.getProperty("key4"));
+
+
+        Map<String, Map<String, DataSourceParameter>> schemaDataSources = proxyConfiguration.getSchemaDataSources();
+        Assert.assertNotNull(schemaDataSources);
+        Assert.assertEquals(1, schemaDataSources.size());
+        Assert.assertTrue(schemaDataSources.containsKey("yamlProxyRule1"));
+        Map<String, DataSourceParameter> dataSourceParameterMap = schemaDataSources.get("yamlProxyRule1");
+        Assert.assertTrue(dataSourceParameterMap.containsKey("ds1"));
+
+        DataSourceParameter dataSourceParameter = dataSourceParameterMap.get("ds1");
+        Assert.assertNotNull(dataSourceParameter);
+        Assert.assertEquals("url1", dataSourceParameter.getUrl());
+        Assert.assertEquals("username1", dataSourceParameter.getUsername());
+        Assert.assertEquals("password1", dataSourceParameter.getPassword());
+        Assert.assertEquals(1, dataSourceParameter.getConnectionTimeoutMilliseconds());
+        Assert.assertEquals(2, dataSourceParameter.getIdleTimeoutMilliseconds());
+        Assert.assertEquals(3, dataSourceParameter.getMaxLifetimeMilliseconds());
+        Assert.assertEquals(4, dataSourceParameter.getMaxPoolSize());
+        Assert.assertEquals(5, dataSourceParameter.getMinPoolSize());
+        Assert.assertEquals(6, dataSourceParameter.getMaintenanceIntervalMilliseconds());
+        Assert.assertTrue(dataSourceParameter.isReadOnly());
+
+        Map<String, Collection<RuleConfiguration>> schemaRules = proxyConfiguration.getSchemaRules();
+        Assert.assertNotNull(schemaRules);
+        Assert.assertEquals(1, schemaRules.size());
+        Collection<RuleConfiguration> ruleConfigurationCollection = schemaRules.get("yamlProxyRule1");
+        Assert.assertNotNull(ruleConfigurationCollection);
+        Assert.assertEquals(1, ruleConfigurationCollection.size());
+        RuleConfiguration ruleConfiguration = ruleConfigurationCollection.iterator().next();
+        Assert.assertNotNull(ruleConfiguration);
+        Assert.assertTrue(ruleConfiguration instanceof MasterSlaveRuleConfiguration);
+    }
+
+}
```

Comment:
should add a new empty line at the end of the code for checkstyle

Reply:
Just formatted with checkstyle.